### PR TITLE
Filter out resources not intended for this framework

### PR DESCRIPTION
**Pull request by gabrielhartmann**
In order to provide a guarantee of safety regarding offer handling we should not even consider resources which are not intended for us.  [The crux of the logic is here](https://github.com/mesosphere/dcos-commons/compare/resource_filter?expand=1#diff-07dc1ba43f35caa9802d00f9ddf66dc0R154).

If no dynamic reservations have been applied then we can feel free to consume the resource.  If reservations have been applied we should be certain it we applied them.

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceUtils.java [hunk 1]
```diff
@@ -132,6 +132,29 @@ public static boolean isOwnedByThisFramework(Resource resource, FrameworkInfo fr
...
+    public static Collection<Resource.ReservationInfo> getDynamicReservations(Resource resource) {
+        return getReservations(resource).stream()
+                .filter(reservationInfo -> reservationInfo.hasType())
+                .filter(reservationInfo -> reservationInfo.getType()
+                        .equals(Protos.Resource.ReservationInfo.Type.DYNAMIC))
+                .collect(Collectors.toSet());
+    }
+
+    public static boolean isProcessable(Protos.Resource resource, Protos.FrameworkInfo frameworkInfo) {
+        boolean hasDynamicReservations = ResourceUtils.getDynamicReservations(resource).size() > 0;
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceUtils.java]: If you wanted to optimize out pulling the dynamic reservations, you could check isOwnedBy first, return if true, then check dynamic reservations.

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/OfferUtilsTest.java [hunk 2]
```diff
@@ -82,6 +85,22 @@ public void testDeclineOffers() {
...
     }
 
+    @Test
+    public void testUnreservedResourcesIsProcessable() {
+        Protos.Resource resource = ResourceTestUtils.getUnreservedCpus(1.0);
+        Assert.assertTrue(ResourceUtils.isProcessable(resource, null));
+    }
+
+    @Test
+    public void testReservedByThisFrameworkIsProcessable() {
```
Comment (benclarkwood) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/OfferUtilsTest.java]: Test for alien resources aren't processable?

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/OfferUtilsTest.java [hunk 3]
```diff
@@ -1,8 +1,10 @@
 package com.mesosphere.sdk.offer;
```
Comment (benclarkwood) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/OfferUtilsTest.java]: I may just not be seeing where you test the alien case of isProcessable or clean in OfferUtils?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceUtils.java (hunk 1 above)
Comment (gabrielhartmann) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceUtils.java]: They're of a similar time complexity.

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/OfferUtilsTest.java (hunk 3 above)
Comment (gabrielhartmann) [reply sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/OfferUtilsTest.java]: https://github.com/mesosphere/dcos-commons/blob/master/sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/ResourceUtilsTest.java#L18-L120

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/OfferUtilsTest.java (hunk 2 above)
Comment (gabrielhartmann) [reply sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/OfferUtilsTest.java]: https://github.com/mesosphere/dcos-commons/blob/master/sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/ResourceUtilsTest.java#L18-L120

**Pull request closed without merging**